cloud/edgecontroller/pkg/controller/manager/location.go

```diff
@@ -2,17 +2,24 @@ package manager
 
 import (
 	"fmt"
+	"strings"
 	"sync"
 
+	"github.com/kubeedge/kubeedge/cloud/edgecontroller/pkg/controller/utils"
+	"github.com/kubeedge/beehive/pkg/common/log"
+
 	"k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // LocationCache cache the map of node, pod, configmap, secret
 type LocationCache struct {
+	//NodeRoleMap is a map, key is node id, value is role
+	NodeRoleMap   sync.Map
```

Comment:
NodeRoleMap -> nodeRoleMap ?

Reply:
using it in controller package also to check for events with nodeid of other role